shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -57,7 +66,45 @@ interface TextureEntry {
     void release();
   }
 
+  interface ImageProducerTextureEntry extends TextureEntry {
+    /**
+     * Retrieve the last Image produced. Drops all previously produced images.
+     *
+     * <p>NOTE: Caller must call Image.close() on returned image.
+     *
+     * @return Image or null.
+     */
+    public Image acquireLatestImage();
+  }
+
+  /** A registry entry for texture updated by rendering into a Surface. */
+  @Keep
+  interface RawSurfaceTextureEntry extends TextureEntry {
```

Comment:
Naming idea: I don't like that we end up with `SurfaceTexture` in the set of nouns here. If I put on my Java and Android hat (with names such as Surface/SurfaceHolder/SurfaceTexture), then to me this feels like a `SurfaceProducer` or maybe a `SurfaceManager`or `RawSurfaceProducer`.

At any rate I would find a way to drop `Texture` from the name.

Reply:
Strongly agree that the current name is terrible.

I considered SurfaceHolderTexture and SurfaceManagerTexture, I like SurfaceProducer.